routes/index.js

```diff
@@ -1,67 +1,114 @@
+  db.query(query)
+  .then(results => {
+    res.send(results.rows)
+  })
 })
 
 router.get('/test-mail', function(req,res) {
-    console.log('GET /test-mail')
-    const msg = {
-      to: 'jasonyu6154@gmail.com',
-      from: 'jasonyu6154@gmail.com',
-      subject: 'Sending with Twilio SendGrid is Fun',
-      text: 'and easy to do anywhere, even with Node.js',
-      html: '<strong>and easy to do anywhere, even with Node.js</strong>',
-    }
-    mailClient
-        .send(msg)
-        .then(() => {
-            console.log('sent test message without errors')
-            res.status(202).send('ok sent')
-        }, error => {
-            console.error(error)
-            if (error.response) {
-                console.error(error.response.body)
-            }
-            res.status(error.code).send(error)
-    })
+  console.log('GET /test-mail')
+  const msg = {
+    to: 'jasonyu6154@gmail.com',
+    from: 'jasonyu6154@gmail.com',
+    subject: 'Sending with Twilio SendGrid is Fun',
+    text: 'and easy to do anywhere, even with Node.js',
+    html: '<strong>and easy to do anywhere, even with Node.js</strong>',
+  }
+  mailClient
+    .send(msg)
+    .then(() => {
+      console.log('sent test message without errors')
+      res.status(202).send('ok sent')
+    }, error => {
+      console.error(error)
+      if (error.response) {
+        console.error(error.response.body)
+      }
+      res.status(error.code).send(error)
+  })
 })
 
 // Route for login/adding new user
 router.post('/api/users', function(req,res){
-    const { firstname, lastname, email } = req.body
-    const secret = uuidv1();
+  const { firstname, lastname, email } = req.body
+  const secret = uuidv4();
 
-    if (!firstname || !lastname || !email) {
-        alert('Input field empty');
-        return res.status(400).send('Input field empty');
-    }
-    const query = {
-        text: 'INSERT INTO users(firstname, lastname, email, secret) VALUES ($1, $2, $3, $4)',
-        values: [firstname, lastname, email, secret],
+  if (!firstname || !lastname || !email) {
+    alert('Input field empty');
+    return res.status(400).send('Input field empty');
+  }
+  
+  // Checks for duplicate
+  const checkdup = {
+    text: 'SELECT COUNT(email) FROM users WHERE email = $1',
+    values: [email.toLowerCase()],
+  }
+  db.query(checkdup)
+  .then(result => {
+    if (result.rows[0].count != 0) {
+      res.status(240)
     }
-    db.query(query)
-    .then(result => {
-        res.send([firstname, lastname, email, secret])
-    })
-    .catch(err => {
-        res.status(400).json(err);
-    })
+  })
+
+  //Insert user data
+  const query = {
+    text: 'INSERT INTO users(firstname, lastname, email, secret) VALUES ($1, $2, $3, $4)',
+    values: [firstname.toLowerCase(), lastname.toLowerCase(), email.toLowerCase(), secret],
+  }
+  db.query(query)
+  .then(result => {
+    res.send([firstname, lastname, email, secret])
+  })
+  .catch(err => {
+    res.status(400).json(err);
```

Comment:
[nit] return a 500 (server error) rather than 400 (bad request) to differentiate from when the input is missing